src/components/HistoryTable/Cell.tsx

```diff
@@ -46,10 +47,51 @@ export const EmptyCell = () => (
   </Cell>
 );
 
-export const LoadingCell = () => (
-  <Cell data-cy="loading-cell">
-    <Skeleton.Avatar active shape="circle" size={statusIconSize} />
-  </Cell>
+interface LoadingCellProps {
+  isHeader?: boolean;
+}
+export const LoadingCell: React.FC<LoadingCellProps> = ({
+  isHeader = false,
+}) => (
+  <>
+    {isHeader ? (
+      <HeaderCell data-cy="loading-header-cell">
+        <Skeleton active title paragraph={false} />
+      </HeaderCell>
+    ) : (
+      <Cell data-cy="loading-cell">
+        <Skeleton.Avatar active shape="circle" size={statusIconSize} />
+      </Cell>
+    )}
+  </>
+);
+
+interface ColumnHeaderCellProps {
+  link: string;
+  trimmedDisplayName: string;
+  fullDisplayName: string;
+}
+export const ColumnHeaderCell: React.FC<ColumnHeaderCellProps> = ({
+  link,
+  trimmedDisplayName,
+  fullDisplayName,
+}) => (
+  <HeaderCell data-cy="header-cell">
+    {trimmedDisplayName === fullDisplayName ? (
+      <StyledRouterLink to={link}>{fullDisplayName}</StyledRouterLink>
+    ) : (
+      <Tooltip
+        align="top"
+        justify="middle"
+        trigger={
+          <StyledRouterLink to={link}>{trimmedDisplayName}</StyledRouterLink>
+        }
+        triggerEvent="hover"
+      >
+        {fullDisplayName}
+      </Tooltip>
+    )}
```

Comment:
We have a ConditionalWrapper component that you can probably use for something like this. https://github.com/evergreen-ci/spruce/blob/main/src/components/ConditionalWrapper/index.tsx